apps/sms/test/unit/compose_test.js

```diff
@@ -1454,43 +1482,94 @@ suite('compose_test.js', function() {
     });
   });
 
-  suite('Attachment pre-send menu', function() {
+  // generate checks for image, video, audio, and unknown button texts
+  ['image', 'video', 'audio', 'others'].forEach(function(type) {
+    suite(type + ' attachment pre-send menu display', function() {
+      setup(function() {
+        this.attachment = mockAttachment({ type: type });
+        Compose.clear();
+        Compose.append(this.attachment);
+        this.attachmentSize = Compose.size;
+        this.sinon.stub(window.URL, 'revokeObjectURL');
+
+        // trigger a click on attachment
+        this.attachment.mNextRender.click();
+      });
+      test('show called with correct options', function() {
+        assert.equal(MockOptionMenu.calls.length, 1);
+        var call = MockOptionMenu.calls[0];
+        assert.equal(call.header, TYPE_DATA[type].name);
+        assert.equal(call.items[0].l10nId, 'view-attachment-' + type);
+        assert.equal(call.items[1].l10nId, 'remove-attachment-' + type);
+        assert.equal(call.items[2].l10nId, 'replace-attachment-' + type);
+        assert.equal(call.items[3].l10nId, 'cancel');
+      });
+    });
+  });
+
+  suite('Attachment pre-send menu methods', function() {
+    var call;
+
     setup(function() {
       this.attachment = mockImgAttachment();
       Compose.clear();
       Compose.append(this.attachment);
       this.attachmentSize = Compose.size;
-      this.sinon.stub(AttachmentMenu, 'open');
-      this.sinon.stub(AttachmentMenu, 'close');
       this.sinon.stub(window.URL, 'revokeObjectURL');
 
       // trigger a click on attachment
       this.attachment.mNextRender.click();
+      call = MockOptionMenu.calls[0];
     });
 
-    test('click opens menu', function() {
-      assert.isTrue(AttachmentMenu.open.called);
+    suite('view', function() {
+      setup(function() {
+        this.sinon.stub(this.attachment, 'view');
+        call.items[0].method();
+      });
+      test('clicking on view calls attachment.view', function() {
+        assert.isTrue(this.attachment.view.called);
```

Comment:
nit: I know it was so before you, but maybe you use the chance and use ```sinon.assert.called``` in the changed suite instead?